### DISCUSSION TITLE
[14.0.0] : How to customize AutoComplete empty message from server side ?

### DISCUSSION THREAD

**Discussion by Stan-Le-Floch** [primefaces/community#2093] [PrimeFaces] [answered]
In Primefaces v14.0.0: the `emptyMessage` property for AutoComplete has been replaced by a variable in client side localization. This is a problem for me as I need to customize this message independently for each autoComplete in my project.
I've found a way to customize it using a mix of JavaScript and EL variable (as seen below), but my solution seems far-fetched for a feature that was previously just a property to edit.

Do you know a better way of doing what I'm looking for? Am I missing something?

Primefaces v13.X :

```
  <p:autoComplete completeMethod="#{frag.actionAutocomplete}"  widgetVar="autoComplete" 
  value="#{frag.selectedValue}"  emptyMessage="#{frag.myCustomEmptyMessage}"/>
```

Primefaces v14.0.0 :
```
  <p:autoComplete completeMethod="#{frag.actionAutocomplete}"  widgetVar="autoComplete" value="#{frag.selectedValue}"/>

      <script>
          jQuery(document).ready(function() {
              PF('autoComplete').emptyMessage = '#{frag.myCustomEmptyMessage}';
          });
      </script>
```

**Comment by melloware** [comment, marked as answer]
i think your solution is a reasonable one?

what about: https://www.primefaces.org/showcase/ui/ajax/lifecycle.xhtml?jfwid=81dde

```xml
<p:autoComplete completeMethod="#{frag.actionAutocomplete}"  widgetVar="autoComplete" value="#{frag.selectedValue}" > 
   <f:attribute name="widgetPostConstruct" value="widget.emptyMessage = '#{frag.myCustomEmptyMessage}';"/>
</p:autoComplete>

**Reply by Stan-Le-Floch** [reply]
I like your solution a lot better, thank you !
